marginaleffects/plot.py

```diff
@@ -0,0 +1,193 @@
+        else:
+            plot_obj.errorbar(x, y, yerr=y_std, fmt='o', color=color, label=label)
+
+    return fig
+
+
+def plot_predictions(
+    model,
+    condition=None,
+    by=False,
+    newdata=None,
+    vcov=True,
+    conf_level=0.95,
+    transform=None,
+    draw=True
+):
+
+    assert not (not by and newdata is not None), "The 'newdata' argument requires a 'by' argument"
+
+    assert (condition is None and by) or (condition is not None and not by), "One of the 'condition' and 'by' arguments must be supplied, but not both"
+
+    if by:
+        if isinstance(by, str):
+            by = [by]
+
+        if newdata is not None:
+            dt = predictions(model, newdata=newdata, conf_level=conf_level, vcov=vcov, transform=transform)
+        else:
+            dt = predictions(model, by=by, conf_level=conf_level, vcov=vcov, transform=transform)
+
+        dt = dt.sort(by[0])
+
+    if condition is not None:
+        built_dt = build_plot(model, condition)
+        dt = predictions(model, newdata=built_dt, conf_level=conf_level, vcov=vcov, transform=transform)
+        if isinstance(condition, str):
+            by = [condition]
+        elif isinstance(condition, list):
+            by = condition
+        elif isinstance(condition, dict):
+            by = list(condition.keys())
+
+    if not draw:
+        return dt
+
+    titles_fontsize = 16
+
+    x_type = get_variable_type(by[0], dt)
+
+    if len(by) == 3:
+        fig, axes = plt.subplots(1, dt.n_unique(subset=[by[2]]))
+        color_i = 0
+        color_dict = {}
+
+        for axes_i, by_2val in enumerate(dt.select(by[2]).unique().to_numpy().flatten()):
+            subplot_dt = dt.filter(pl.col(by[2])==by_2val)
+
+            for by_1val in subplot_dt.select(by[1]).unique().to_numpy().flatten():
+                if by_1val not in color_dict:
+                    color_dict[by_1val] = plt.rcParams['axes.prop_cycle'].by_key()['color'][color_i]
+                    color_i += 1
+
+                color_dt = subplot_dt.filter(pl.col(by[1])==by_1val)
+
+                common_plot(color_dt, by[0], x_type, fig=fig, axes_i=axes_i, label=by_1val, color=color_dict[by_1val])
+
+
+            fig.axes[axes_i].set_title(by_2val, fontsize=titles_fontsize)
+
+        legend_elements = [Line2D([0], [0], color=val, label=key) for key,val in color_dict.items()]
+        fig.axes[axes_i].legend(handles=legend_elements, loc='center left', bbox_to_anchor=(1, 0.5), title=by[1], fontsize=titles_fontsize, title_fontsize=titles_fontsize)
+
+    elif len(by) == 2:
+        fig = plt.figure()
+        ax = plt.subplot(111)
+
+        for by_1val in dt.select(by[1]).unique().to_numpy().flatten():
+            color_dt = dt.filter(pl.col(by[1])==by_1val)
+
+            common_plot(color_dt, by[0], x_type, fig=fig, label=by_1val)
+
+        ax.legend(loc='center left', bbox_to_anchor=(1, 0.5), title=by[1], fontsize=titles_fontsize, title_fontsize=titles_fontsize)
```

Comment:
I tried a couple plots and sometimes the legend seems to go out of bounds of the plot. But this may be because of my setup, which is weird (ubuntu via WSL on windows).

Reply:
I'm not sure what you mean. Like this ?

![Screenshot from 2023-08-30 06-33-33](https://github.com/vincentarelbundock/pymarginaleffects/assets/102169875/8fc902d0-8e7f-4fbc-91f8-69a29a917302)

Reply:
Yes

Reply:
Does it displays correctly when the window is expanded?
Do you want me to shrink the plot so that everything appears in the small window?

Reply:
Can't check now, but this is low priority. Let's keep things as is for now.